c_api/docs/taichi/taichi_core.h.md

```diff
@@ -0,0 +1,410 @@
+- `structure.nd_shape.dims`: Dimension sizes.
+
+`structure.nd_array`
+
+Multi-dimentional array of dense primitive data.
+
+- `structure.nd_array.memory`: Memory bound to the ND-array.
+- `structure.nd_array.shape`: Shape of the ND-array.
+- `structure.nd_array.elem_shape`: Shape of the ND-array elements. You usually need to set this if it's a vector or matrix ND-array.
+- `structure.nd_array.elem_type`: Primitive data type of the ND-array elements.
+
+`union.argument_value`
+
+A scalar or structured argument value.
+
+- `union.argument_value.i32`: Value of a 32-bit one's complement signed integer.
+- `union.argument_value.f32`: Value of a 32-bit IEEE 754 floating-poing number.
+- `union.argument_value.ndarray`: An ND-array to be bound.
+
+`structure.argument`
+
+An argument value to feed kernels.
+
+- `structure.argument.type`: Type of the argument.
+- `structure.argument.value`: Value of the argument.
+
+`structure.named_argument`
+
+An named argument value to feed compute graphcs.
+
+- `structure.named_argument.name`: Name of the argument.
+- `structure.named_argument.argument`: Argument body.
+
+`function.create_runtime`
+
+Create a Taichi Runtime with the specified `enumeration.arch`.
+
+`function.destroy_runtime`
+
+Destroy a Taichi Runtime.
+
+`function.allocate_memory`
+
+Allocate a contiguous on-device memory with provided parameters.
+
+`function.free_memory`
+
+Free a memory allocation.
+
+`function.map_memory`
+
+Map an on-device memory to a host-addressible space. The user MUST ensure the device is not being used by any device command before the map.
+
+`function.unmap_memory`
+
+Unmap an on-device memory and make any host-side changes about the memory visible to the device. The user MUST ensure there is no further access to the previously mapped host-addressible space.
+
+`function.create_event`
+
+Create an event primitive.
+
+`function.destroy_event`
+
+Destroy an event primitive.
+
+`function.copy_memory_device_to_device`
+
+Copy the content of a contiguous subsection of on-device memory to another. The two subsections MUST NOT overlap.
+
+`function.launch_kernel`
+
+Launch a Taichi kernel with provided arguments. The arguments MUST have the same count and types in the same order as in the source code.
+
+`function.launch_compute_graph`
+
+Launch a Taichi compute graph with provided named arguments. The named arguments MUST have the same count, names and types as in the source code.
+
+`function.signal_event`
+
+Set an event primitive to a signaled state, so the queues waiting upon the event can go on execution. If the event has been signaled before, the event MUST be reset with `function.reset_event`; otherwise it is an undefined behavior.
+
+`function.reset_event`
+
+Set a signaled event primitive back to an unsignaled state.
+
+`function.wait_event`
+
+Wait on an event primitive until it transitions to a signaled state. The user MUST signal the awaited event; otherwise it is an undefined behavior.
```

Comment:
```suggestion
Waits until an event primitive transitions to a signaled state. You *must* signal the awaited event; otherwise, an undefined behavior would occur.
```
You *must* signal the awaited event. --> Signal the awaited event before calling this method??